java/server/src/org/openqa/selenium/grid/distributor/DrainNode.java

```diff
@@ -0,0 +1,43 @@
+package org.openqa.selenium.grid.distributor;
+
+import static org.openqa.selenium.remote.http.Contents.asJson;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.openqa.selenium.grid.data.NodeId;
+import org.openqa.selenium.remote.http.HttpHandler;
+import org.openqa.selenium.remote.http.HttpRequest;
+import org.openqa.selenium.remote.http.HttpResponse;
+
+import java.io.UncheckedIOException;
+import java.util.Objects;
+import java.util.UUID;
+
+public class DrainNode implements HttpHandler {
+
+  private final Distributor distributor;
+  private final NodeId nodeId;
+
+  public DrainNode(Distributor distributor, NodeId nodeId) {
+    this.distributor = Objects.requireNonNull(distributor);
+    this.nodeId = Objects.requireNonNull(nodeId);
+  }
+
+  @Override
+  public HttpResponse execute(HttpRequest req) throws UncheckedIOException {
+    HttpResponse response = new HttpResponse();
+    boolean value = distributor.drain(nodeId);
+
+    if (value) {
+      response.setContent(
+          asJson(ImmutableMap.of("value", value, "message",
```

Comment:
Style nit: prefer adding the newline _before_ adding the parameters, and the group parameters meaningfully, or "one per line" if there's no rhyme or reason to them.

```
asJson(ImmutableMap.of(
    "value", value,
    "message", "....")));
```